carpics_360v0.1.js

```diff
@@ -625,24 +787,452 @@ var CarPicsSpinnerAPI = (function() {
             this.HTMLElement.style.top = 0 + 'px';
         }
         /*
-         * Allows moving view within zoomed element.
-         */
-        this.move = function(mouseX, mouseY) {
-            var offset = this.HTMLElement.parentElement.getBoundingClientRect();
-            var mouseXOffset;
-            var mouseYOffset;
-            if (mouseX > offset.left && mouseX < (offset.left + offset.width) && mouseY > offset.top && mouseY < (offset.top + offset.height)) {
-                mouseXOffset = offset.left - mouseX;
-                mouseYOffset = offset.top - mouseY;
+        * Allows moving view within zoomed element.
+        */
+        this.move = function(baseEvent, offset, zoomed) {
+            var multiplier = zoomed?1:2;
+            var parentOffset = this.HTMLElement.parentElement.getBoundingClientRect();
+            var clientX = (baseEvent.type !== "touchmove" ? baseEvent.clientX : baseEvent.targetTouches[0].clientX) - parentOffset.left;
+            var clientY = (baseEvent.type !== "touchmove" ? baseEvent.clientY : baseEvent.targetTouches[0].clientY) - parentOffset.top;
+            var correctX =  (-clientX + offset.left - parentOffset.left)*multiplier + parentOffset.width/2;
+            var correctY =  (-clientY + offset.top - parentOffset.top)*multiplier + parentOffset.height/2;
+            if(correctX>0){
+                correctX=0;
+            } else if (-correctX > (multiplier*offset.width-parentOffset.width)) {
+                correctX = -(multiplier*offset.width-parentOffset.width);
+            }
+            if(correctY>0){
+                correctY=0;
+            } else if (-correctY > (multiplier*offset.height-parentOffset.height)) {
+                correctY = -(multiplier*offset.height-parentOffset.height);
+            }
+            this.HTMLElement.style.left = correctX + 'px';
+            this.HTMLElement.style.top = correctY + 'px';
+        }
+        /**
+        * Allows drag to scroll image within zoomed element
+        */
+        this.dragMove = function(baseEvent, offset, zoomed) {
+            var parentOffset = this.HTMLElement.parentElement.getBoundingClientRect();
```

Comment:
We should have a configuration toggle for inertial move vs exact move.  Will discuss tomorrow.